### ISSUE TITLE
Add statistics to result

### ISSUE THREAD

**Issue by thekid** [xp-framework/unittest#33]
It would be great if tools like [xp-forge/coverage](https://github.com/xp-forge/coverage/) could add statistics to the unittest output, like so:

![image](https://user-images.githubusercontent.com/696742/45598297-effa4c00-b9d9-11e8-805d-df506dc7af42.png)

However, the `testRunFinished()` method of a listener is invoked prior to the statistics being printed. This way, output generated there is printed somewhere inbetween:

![image](https://user-images.githubusercontent.com/696742/45598314-29cb5280-b9da-11e8-9c72-342264b3e382.png)

Maybe the best solution would be to enable listeners to add a statistic to `unittest.TestResult`; which is passed as argument.

**Comment by thekid** [comment]
Example:

```php
class CoverageListener implements TestListener {

  // ...

  public function testRunFinished(TestSuite $suite, TestResult $result) {
    $this->coverage->stop();
    $result->metric('Coverage', sprintf('▲ %.2f%% lines covered (%d/%d)', ...));
  }
}

**Comment by thekid** [comment]
If we only register a string for printing, there's no way to access the raw value, which we might want to do inside a test run loop, displaying trends.

Maybe like this?

```php
$result->metric('Coverage', new Metric('%.2f%% lines covered (%d/%d)', function() {
  $report= $this->coverage->getReport();
  $percent= $report->getNumExecutedLines() / $report->getNumExecutableLines() * 100;
  return [$percent, $report->getNumExecutedLines(), $report->getNumExecutableLines()];
}));
```

**Comment by thekid** [comment]
Released in [9.7.0](https://github.com/xp-framework/unittest/releases/tag/v9.7.0)
